pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -156,66 +156,62 @@ func (ps *PipelineRunSpec) validateInlineParameters(ctx context.Context) (errs *
 	if ps.PipelineSpec == nil {
 		return errs
 	}
-	var paramSpec []ParamSpec
+	paramSpecForValidation := make(map[string]ParamSpec)
 	for _, p := range ps.Params {
-		pSpec := ParamSpec{
-			Name:    p.Name,
-			Default: &p.Value,
+		paramSpecForValidation = createParamSpecFromParam(p, paramSpecForValidation)
+	}
+	for _, p := range ps.PipelineSpec.Params {
+		var err *apis.FieldError
+		paramSpecForValidation, err = combineParamSpec(p, paramSpecForValidation)
+		if err != nil {
+			errs = errs.Also(err)
 		}
-		paramSpec = append(paramSpec, pSpec)
 	}
-	paramSpec = appendParamSpec(paramSpec, ps.PipelineSpec.Params)
 	for _, pt := range ps.PipelineSpec.Tasks {
-		paramSpec = appendParam(paramSpec, pt.Params)
+		paramSpecForValidation = appendPipelineTaskParams(paramSpecForValidation, pt.Params)
 		if pt.TaskSpec != nil && pt.TaskSpec.Params != nil {
-			paramSpec = appendParamSpec(paramSpec, pt.TaskSpec.Params)
+			for _, p := range pt.TaskSpec.Params {
+				var err *apis.FieldError
+				paramSpecForValidation, err = combineParamSpec(p, paramSpecForValidation)
+				if err != nil {
+					errs = errs.Also(err)
+				}
+			}
 		}
 	}
+	var paramSpec []ParamSpec
+	for _, v := range paramSpecForValidation {
+		paramSpec = append(paramSpec, v)
+	}
 	if ps.PipelineSpec != nil && ps.PipelineSpec.Tasks != nil {
 		for _, pt := range ps.PipelineSpec.Tasks {
 			if pt.TaskSpec != nil && pt.TaskSpec.Steps != nil {
+				errs = errs.Also(ValidateParameterTypes(ctx, paramSpec))
```

Comment:
@jerop this is the line that caused the array test to fail and so I needed to add the Type field.